lib/python/rosie/svn_pre_commit.py

```diff
@@ -290,19 +289,30 @@ def run(self, repos, txn):
 
             # Admin users and those in access list can modify everything in
             # trunk apart from specific entries in the trunk info file
-            if "*" in access_list or author in admin_users + access_list:
+            if ("*" in access_list or author in admin_users + access_list
+                    or author_aliases.get(author) in access_list):
                 if path_tail != self.TRUNK_INFO_FILE:
                     continue
             else:
                 msg = "User not in access list"
                 bad_changes.append(BadChange(status, path, content=msg))
                 continue
 
-            # Only the admin users can change owner and access list
+            # Only the admin users can change owner and access list except
+            # that access list users can change them to aliases.
+
             if owner == txn_owner and access_list == txn_access_list:
+                # No substantive change made, fine.
                 continue
+
+            if (owner == author_aliases.get(txn_owner)
+                    and set(access_list)
+                        == set([author_aliases.get(_) for _ in txn_access_list])):
```

Comment:
Unnecessary list
```suggestion
            if (owner == author_aliases.get(txn_owner)
                    and set(access_list)
                        == set(author_aliases.get(_) for _ in txn_access_list)):
```
or could use `symmetric_difference()` instead
```suggestion
            if (
                owner == author_aliases.get(txn_owner)
                and not set(access_list).symmetric_difference(
                    author_aliases.get(_) for _ in txn_access_list
                )
            ):
```